### PR TITLE
feat: add support and tests for struct fields

### LINKED ISSUE #21: BigQuery: Upload STRUCT / RECORD fields from load_table_from_dataframe
**Is your feature request related to a problem? Please describe.**

If you have a pandas Series containing dictionaries, ideally this could be uploaded to BigQuery as a STRUCT / RECORD column. Currently this fails with a "file does not exist error" because the arrow write_table fails with ""ArrowInvalid: Nested column branch had multiple children".

**Describe the solution you'd like**

Upload of a RECORD column succeeds. This will require a fix to https://jira.apache.org/jira/browse/ARROW-2587.

**Describe alternatives you've considered**

Change intermediate file format to JSON or some other type. This isn't ideal, since most other types are row-oriented, but pandas DataFrames are column-oriented.

### PR DESCRIPTION
**Pull request by HemangChothani**
Fixes #21 

- Pyarrow version 0.17.0 has released so issue has been resolved but it doesn't support `python2` so add condition for `valueerror` which raised when python2 excecuted.

### CODE REVIEW

**Comment by plamut** [comment]
Just a thought - if I understand correctly, this fix will only work in Python 3 due to the new `pyarrow`'s incompatibility with Python 2?

If so, we might want to wait with this for a bit, since a green light was given to start transitioning the libraries to the [new code generator](https://github.com/googleapis/gapic-generator-python), which also implies **dropping Python 2 support**.

Before this transition one final Python 2 compatible release will be made, but since this fix only affects Python 3, we might want to wait for an extra week or two, and then merge a simplified version of it.

**Comment by plamut** [comment]
It turned out there is still some work needed to transition the library to the new microgenerator (e.g. transitioning the dependencies first), which might take a while. Considering that, it's probably not worth waiting for it and getting the fix in sooner has value.

I'll review this tomorrow and see if it's good to merge and include in the next (non-major) release.

File: google/cloud/bigquery/_pandas_helpers.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
 import functools
 import logging
 import warnings
-
```
Comment (tseaver) [inline google/cloud/bigquery/_pandas_helpers.py:21]: This empty line divides standard library imports from third party imports, and should stay.

File: google/cloud/bigquery/_pandas_helpers.py [hunk 2]
```diff
@@ -18,8 +18,8 @@
...
 import six
+
```
Comment (tseaver) [inline google/cloud/bigquery/_pandas_helpers.py:22]: This one should go.

File: google/cloud/bigquery/_pandas_helpers.py [hunk 3]
```diff
@@ -287,13 +287,14 @@ def dataframe_to_bq_schema(dataframe, bq_schema):
...
-                    "Uploading dataframes with struct (record) column types "
-                    "is not supported. See: "
-                    "https://github.com/googleapis/google-cloud-python/issues/8191"
-                )
+        if six.PY2:
+            for field in bq_schema:
+                if field.field_type in schema._STRUCT_TYPES:
+                    raise ValueError(
+                        "Uploading dataframes with struct (record) column types "
+                        "is not supported. See: "
```
Comment (tseaver) [inline google/cloud/bigquery/_pandas_helpers.py:295]: Should this line indicate the Python2-only part, e.g.:
```suggestion
                        "is not supported under Python2. See: "
```

File: google/cloud/bigquery/_pandas_helpers.py [hunk 4]
```diff
@@ -287,13 +287,14 @@ def dataframe_to_bq_schema(dataframe, bq_schema):
...
+                        "https://github.com/googleapis/google-cloud-python/issues/8191"
```
Comment (tseaver) [inline google/cloud/bigquery/_pandas_helpers.py:296]:
```suggestion
                        "https://github.com/googleapis/python-bigquery/issues/21"
```

**Comment by tseaver** [comment]
@shollyman, @plamut I can't see any log for the failed `Kokoro docs-presubmit` job.  Is that a tooling thing?

**Comment by tseaver** [comment]
On chat, I see that the `Kokoro docs presubmit` failures are known / expected while the infrastructure / configs get rolled out.

**Comment by tseaver** [comment]
@HemangChothani It looks like you have turned off commits from mainainers for this PR's branch, which means the automerge bot cannot "push the button" for you (nor can I).

**Comment by HemangChothani** [comment]
@tseaver I thnik that's my bad, can i merge this PR.

**Comment by plamut** [comment]
@HemangChothani In principle yes, but in the meantime @busunkim96 closed all PRs. It might be because of the CI maintenance work or something, don't know yet.

**Pull request merged**